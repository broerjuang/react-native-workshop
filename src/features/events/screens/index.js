//@flow

export {default as EventsScreen} from './EventsScreen';
