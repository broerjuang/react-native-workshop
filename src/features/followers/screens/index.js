// @flow
export {default as FollowersScreen} from './FollowersScreen.js';
