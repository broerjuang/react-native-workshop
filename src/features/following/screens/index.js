// @flow
export {default as FollowingScreen} from './FollowingScreen.js';
