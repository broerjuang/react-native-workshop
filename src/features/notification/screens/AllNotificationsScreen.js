// @flow

import React, {Component} from 'react';
import {View, ScrollView, Text, TouchableOpacity} from 'react-native';
import {Avatar, ButtonGroup} from 'react-native-elements';
import {SafeAreaView} from 'react-navigation';
import {connect} from 'react-redux';
import Button from '../../../global/core-ui/Button';
import RowWith3Column from '../../../global/core-ui/RowWith3Column';
import {Ionicons, Octicons} from '@expo/vector-icons';

type Props = {
  markAsRead: (selectedNotificationID: number) => void;
  markAllAsRead: () => void;
  handleAction: (action: Object) => void;
  allNotificationData: Array<*>;
  unreadNotificationData: Array<*>;
  participatingNotificationData: Array<*>;
};

type NotifListType = {
  id: number;
  reposID: number;
  reposFullName: string;
  reposOwner: string;
  reposAvatar: string;
  subjectTitle: string;
  subjectURL: toString;
  subjectType: string;
  isUnread: boolean;
  isFork: boolean;
};
type State = {
  unreadNotifList: Array<NotifListType>;
  readNotifList: Array<NotifListType>;
  participatingNotifList: Array<NotifListType>;
  selectedIndex: number;
};

export class AllNotificationsScreen extends Component<Props, State> {
  state = {
    unreadNotifList: [],
    readNotifList: [],
    participatingNotifList: [],
    selectedIndex: 0,
  };

  async componentDidMount() {
    this.props.handleAction({type: 'NOTIFICATION_REQUESTED'});
  }

  updateIndex(selectedIndex: number) {
    this.setState({selectedIndex});
  }

  render() {
    let notificationDataForMap: Array<*> = [];

    if (this.state.selectedIndex === 0) {
      notificationDataForMap = this.props.unreadNotificationData;
    }
    if (this.state.selectedIndex === 1) {
      notificationDataForMap = this.props.participatingNotificationData;
    }
    if (this.state.selectedIndex === 2) {
      notificationDataForMap = this.props.allNotificationData;
    }

    var temporaryStoreForData = {};

    notificationDataForMap.forEach((item) => {
      if (
        typeof temporaryStoreForData[item.repository.full_name] === 'undefined'
      ) {
        temporaryStoreForData[item.repository.full_name] = [];
      }
      temporaryStoreForData[item.repository.full_name].push({
        id: item.id,
        subjectTitle: item.subject.title,
        subjectURL: item.subject.url,
        subjectType: item.subject.type,
        isUnread: item.unread,
        reposFullName: item.repository.full_name,
        reposAvatar: item.repository.owner.avatar_url,
      });
    });

    var newData = [];

    for (var reposID in temporaryStoreForData) {
      if (reposID) {
        newData.push({
          reposFullName: reposID,
          notifications: [],
        });

        var lastItem = newData.length - 1;

        temporaryStoreForData[reposID].forEach((item) => {
          newData[lastItem].notifications.push(item);
        });
      }
    }

    const buttons = ['Unread', 'Participating', 'All'];

    let showNotificationList;

    if (notificationDataForMap.length === 0) {
      showNotificationList = (
        <Text style={noNotificationStyle}>
          {`You don't have notifications of this type.`}
        </Text>
      );
    }

    return (
      <View style={mainContainer}>
        <SafeAreaView />
        <ButtonGroup
          buttons={buttons}
          onPress={this.updateIndex.bind(this)}
          selectedIndex={this.state.selectedIndex}
        />

        <ScrollView>
          {this.state.selectedIndex === 0 ? (
            <Button
              style={{
                flex: 1,
                shadowOffset: {width: 1, height: 1},
                shadowColor: 'black',
                shadowOpacity: 0.2,
                borderColor: 'black',
              }}
              backgroundColor={'#F1F1F1'}
              title={'MARK ALL AS READ'}
              underlayColor={'#E7E7E7'}
              onPress={this.props.markAllAsRead}
            />
          ) : (
            ''
          )}

          <View>
            {newData.map((item, i) => (
              <View style={mainList} key={i}>
                <RowWith3Column
                  style={{backgroundColor: '#F1F1F1'}}
                  left={
                    <Avatar
                      small
                      rounded
                      source={{uri: item.notifications[0].reposAvatar}}
                    />
                  }
                  content={<Text>{item.reposFullName}</Text>}
                />

                {item.notifications.map((notif, i) => (
                  <RowWith3Column
                    style={{
                      borderBottomColor: '#F1F1F1',
                      borderBottomWidth: 2,
                    }}
                    key={i}
                    left={
                      notif.subjectType === 'PullRequest' ? (
                        <Octicons
                          name="git-pull-request"
                          size={25}
                          type="OCTICONS"
                        />
                      ) : (
                        <Octicons
                          name="issue-opened"
                          size={25}
                          type="OCTICONS"
                        />
                      )
                    }
                    content={<Text>{notif.subjectTitle}</Text>}
                    right={
                      notif.isUnread ? (
                        <TouchableOpacity
                          onPress={() => this.props.markAsRead(notif.id)}
                        >
                          <Ionicons
                            name="ios-checkmark"
                            size={50}
                            color="black"
                          />
                        </TouchableOpacity>
                      ) : (
                        ''
                      )
                    }
                  />
                ))}
              </View>
            ))}
            {showNotificationList}
          </View>
        </ScrollView>
      </View>
    );
  }
}

const mainContainer = {
  flex: 1,
};

const mainList = {
  flex: 1,
  margin: 10,
  shadowOffset: {width: 1, height: 1},
  shadowColor: 'black',
  shadowOpacity: 0.2,
};

const noNotificationStyle = {
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  flexDirection: 'row',
};

function mapStateToProps(state) {
  return {
    unreadNotificationData: state.notificationReducer.unreadNotificationData,
    participatingNotificationData:
      state.notificationReducer.participatingNotificationData,
    allNotificationData: state.notificationReducer.allNotificationData,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    handleAction: (action: Object) => dispatch(action),
    markAllAsRead: () => {
      dispatch({
        type: 'MARK_ALL_NOTIFICATIONS_AS_READ',
      });
    },
    markAsRead: (selectedNotificationID: string) => {
      return dispatch({
        type: 'MARK_NOTIFICATION_AS_READ',
        payload: selectedNotificationID,
      });
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AllNotificationsScreen);
