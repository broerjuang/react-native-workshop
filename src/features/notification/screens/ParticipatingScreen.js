// @flow
import React, {Component} from 'react';
import {View, Text, StyleSheet} from 'react-native';

export default class ParticipatingScreen extends Component<{}> {
  render() {
    const {noNotifText, mainContainer} = styles;

    return (
      <View style={mainContainer}>
        <Text style={noNotifText}>
          You dont have any notifications of this type.
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: 'white',
  },
  noNotifText: {
    marginTop: 10,
    textAlign: 'center',
    alignSelf: 'center',
  },
  mainList: {
    height: 100,
    margin: 10,
    shadowOffset: {width: 1, height: 1},
    shadowColor: 'black',
    shadowOpacity: 0.2,
  },
  upperListPart: {
    flex: 1,
    backgroundColor: '#F1F1F1',
  },
  listDetails: {
    flexDirection: 'row',
    margin: 10,
  },
  listText: {
    marginLeft: 5,
    alignSelf: 'center',
    fontWeight: 'bold',
  },
  bottomListPart: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: 'white',
    paddingLeft: 5,
  },
  bottomDetailsText: {
    textAlign: 'left',
    alignSelf: 'center',
    padding: 5,
    flexWrap: 'wrap',
    fontSize: 12,
  },
});
