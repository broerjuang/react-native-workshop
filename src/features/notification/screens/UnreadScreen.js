// @flow

import React, {Component} from 'react';
import {View, Text} from 'react-native';

class UnreadScreen extends Component<{}> {
  render() {
    return (
      <View>
        <Text>Unread Screen</Text>
      </View>
    );
  }
}

export default UnreadScreen;
