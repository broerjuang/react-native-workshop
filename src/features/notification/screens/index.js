//@flow
export {default as AllNotificationsScreen} from './AllNotificationsScreen';
export {default as ParticipatingScreen} from './ParticipatingScreen';
export {default as UnreadScreen} from './UnreadScreen';
