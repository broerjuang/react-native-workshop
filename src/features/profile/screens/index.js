export {default as ProfileScreen} from './ProfileScreen';
export {default as SettingScreen} from './SettingScreen';
