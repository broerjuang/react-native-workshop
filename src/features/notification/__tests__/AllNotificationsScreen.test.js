// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import {AllNotificationsScreen} from '../screens/AllNotificationsScreen';

describe('AllNotificationsScreen', () => {
  it('should render AllNotificationsScreen corectly', () => {
    let component = renderer.create(
      <AllNotificationsScreen
        navigation={{navigate: (s) => s}}
        markAsRead={(s: number) => {
          return;
        }}
        markAllAsRead={() => {
          return;
        }}
        handleAction={(action: Object) => {
          return;
        }}
        allNotificationData={[]}
        unreadNotificationData={[]}
        participatingNotificationData={[]}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
