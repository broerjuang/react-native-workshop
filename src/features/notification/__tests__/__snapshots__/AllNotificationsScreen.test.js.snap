// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllNotificationsScreen should render AllNotificationsScreen corectly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={undefined}
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderColor": "#e3e3e3",
          "borderRadius": 3,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "marginBottom": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 5,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 0,
          },
          false,
          false,
          Object {
            "borderBottomLeftRadius": 3,
            "borderTopLeftRadius": 3,
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                },
                undefined,
                Object {
                  "color": "#43484d",
                },
                undefined,
              ],
            ]
          }
        >
          Unread
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 1,
          },
          Object {
            "borderLeftColor": "#bdc6cf",
            "borderLeftWidth": 1,
          },
          false,
          false,
          false,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                },
                undefined,
                false,
                false,
              ],
            ]
          }
        >
          Participating
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          false,
          false,
          Object {
            "borderBottomRightRadius": 3,
            "borderTopRightRadius": 3,
          },
          false,
          false,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                },
                undefined,
                false,
                false,
              ],
            ]
          }
        >
          All
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            Object {
              "borderRadius": 3,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "black",
              "flex": 1,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.2,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#F1F1F1",
                },
                Object {
                  "borderRadius": 3,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "black",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    null,
                    undefined,
                  ],
                ]
              }
            >
              MARK ALL AS READ
            </Text>
          </View>
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          You don't have notifications of this type.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
